suzieq/poller/worker/nodes/node.py

```diff
@@ -1187,6 +1188,93 @@ async def rest_gather(self, service_callback, cmd_list, cb_token,
 class IosXENode(Node):
     '''IOS-XE Node-sepcific telemetry gather implementation'''
 
+    async def _init_ssh(self, init_boot_time=True, rel_lock: bool = True) -> None:
+        '''Need to start an interactive session for XE
+
+        Many IOSXE devices cannot accept commands as fast as we can fire them.
+        Furthermore, many TACACS servers also collapse under the load of
+        multiple authemtication requests because each SSH session needs to be
+        authenticated, not just the main SSH connection.
+        '''
+        backoff_period = 1
+        self.WAITFOR = r'.*[>#]\s*$'
+
+        while not self._conn:
+            await super()._init_ssh(init_boot_time=False, rel_lock=False)
+            if not self._conn:
+                await asyncio.sleep(backoff_period)
+                backoff_period *= 2
+                backoff_period = min(backoff_period, 120)
+                if rel_lock:
+                    self.ssh_ready.release()
+                return
```

Comment:
I'm not sure I understood what you are doing here. Why are you setting the backoff_period if you are going to return?That value will be lost. 
Moreover, that return makes the `while not self._conn` useless because you will never trigger it more than 1 time.

Another comment is related to the fact that you are releasing the lock after the backoff_period. What is the reason to keep the lock while it is sleeping? To slow down the commands?

Reply:
Ugh. Fixing.